### PR TITLE
fix(vercel): Implement pagination for get env var endpoint

### PR DESCRIPTION
**Pull request by ceorourke**
If the first page (20 results) of environment variables that we fetch does not include the Sentry/Vercel env vars that we look for, `env_var_already_exists` will return False and we'll attempt to create the environment variable, only to find out upon creation that it does actually exist (but our env var is in another castle). The solution is to paginate to get all (well, 200) the env vars and then check if it already exists or not.

Fixes [SENTRY-N4D](https://sentry.io/organizations/sentry/issues/2201700901/?query=ENV_ALREADY_EXISTS&statsPeriod=14d), [SENTRY-N4G](https://sentry.io/organizations/sentry/issues/2201989893/?query=ENV_ALREADY_EXISTS&statsPeriod=14d), and [SENTRY-JW3](https://sentry.io/organizations/sentry/issues/2069926640/?query=ENV_ALREADY_EXISTS&statsPeriod=14d).

### CODE REVIEW

**Comment by ceorourke** [comment]
It's also worth noting (though outside the scope of this PR) that an [update an env var endpoint](https://vercel.com/docs/api#endpoints/projects/edit-a-project-environment-variable) now exists (and didn't when we created this, as well as get and create being on the same version now as opposed to us using v4 to get and v5 to post (that's what the docs had at the time of creation though).

File: tests/sentry/integrations/vercel/test_integration.py [hunk 1]
```diff
@@ -205,7 +205,10 @@ def test_update_organization_config(self):
             responses.GET,
             "https://api.vercel.com/v5/projects/%s/env"
             % "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H",
-            json={"envs": []},
+            json={
+                "envs": [],
+                "pagination": {"count": 10},
```
Comment (MeredithAnya) [inline tests/sentry/integrations/vercel/test_integration.py:210]: do we actually get the pagination if we don't have results?

Comment (ceorourke) [reply tests/sentry/integrations/vercel/test_integration.py:210]: like is pagination in the response if there are no env vars? I'll check

**Review by MeredithAnya — APPROVED** [review]
I think I would just make the test response consistent with what we are getting so if `envs` is empty then pagination is `0` and similar thing for the other test response

**Pull request merged**